changelog.txt

```diff
@@ -1,5 +1,6 @@
 V.Next
 ---------
+- [PATCH] Move MS STS Response handling to separate class (#2471)
```

Comment:
This is certainly not a patch, in fact not even a minor. We are making breaking changes to `public` interfaces by changing their method signatures. If we proceed with the change currently in the PR, then this needs to be marked as MAJOR

Reply:
Yes, I will update this as well per final changes in this PR.

Reply:
This PR not introducing breaking public API change

Reply:
I think what @shahzaibj meant is that this is a breaking change for common

Reply:
@shahzaibj from what I understand, @mohitc1 only extracted logic from MicrosoftStsOnAuth2Strategy here.

MicrosoftStsOnAuth2Strategy's interface remains untouched = no breaking change.

Reply:
Yes, this PR is no more changing that interface. First iteration did, I will introduce that in next PR.